2Dgame/Assets/Scripts/PlatformSpawnerScript.cs

```diff
@@ -40,83 +45,81 @@ public class PlatformSpawnerScript : MonoBehaviour {
 	 * 		10% chance trampoline
 	 * 		10% chance moving
 	 */ 
-	GameObject basicPlatformScheme() {
-        return PlatformScheme(80, 10, 10);
+	void basicPlatformScheme() {
+        PlatformScheme(80, 10, 10);
 	}
 
     /**
      * Platformscheme with a variable amount of parameters.
      * 
      * Returns a platform prefab.
      */
-    GameObject PlatformScheme(params int[] percentages)
+    void PlatformScheme(params int[] percentages)
     {
-        if (didWeFlip)
-        {
-            flipCounter++;
-            if (flipCounter % 6 == 0)
-            {
-                didWeFlip = false;
-                return platformPrefabs[3];
-            }
-        }
-
         int randomInt = Random.Range(0, 100);
         int percentageCounter = 0;
         for (int i = 0; i < percentages.Length; i++)
         {
             percentageCounter += percentages[i];
             if (randomInt < percentageCounter)
             {
+                prefabqueue.Enqueue(platformPrefabs[i]);
                 if (i == 3)
                 {
-                    if (didWeFlip)
+                    for (int j = 0; j < 3; j++)
                     {
-                        flipCounter = 0;
+                        prefabqueue.Enqueue(platformPrefabs[1]);
                     }
-                    else
+                    prefabqueue.Enqueue(platformPrefabs[3]);
+                }
+                else if (i == 4)
+                {
+                    for (int j = 0; j < 3; j++)
                     {
-                        flipCounter++;
+                        prefabqueue.Enqueue(platformPrefabs[5]);
                     }
-					didWeFlip = !didWeFlip;
                 }
-
-                return platformPrefabs[i];
+                return;
             }
         }
-        return platformPrefabs[0];
     }
 
     /**
      * Method for choosing platformscheme.
      * Is the game's primary progressionsystem.
      * Difficulty increases as score increases.
      */
-    GameObject choosePlatform()
+    void choosePlatform()
     {
         if (this.transform.position.y < 200)
         {
-            return basicPlatformScheme();
+            basicPlatformScheme();
+            return;
```

Comment:
You are correct. For a minute there I thought the bug I had was that it fell through the if statements.